### ISSUE TITLE
How to train model in freqai only on marked rows?

### ISSUE THREAD

**Issue by fridary** [freqtrade/freqtrade#9807]
* Python Version: 3.11
  * Freqtrade Version: 2024.1
  
I want to try to train model only on selected rows, for example, where I made entry signal.
Let's take SampleStrategy:
```python
dataframe.loc[
            (
                # Signal: RSI crosses above 30
                (qtpylib.crossed_above(dataframe['rsi'], self.buy_rsi.value)) &
                (dataframe['tema'] <= dataframe['bb_middleband']) &  # Guard: tema below BB middle
                (dataframe['tema'] > dataframe['tema'].shift(1)) &  # Guard: tema is raising
                (dataframe['volume'] > 0)  # Make sure Volume is not 0
            ),
            'enter_long_marked'] = 1
```

How do I execute script to train model (xgboost/NN) only on `enter_long_marked` rows?

**Comment by xmatthias** [comment]
I do assume that what you mean is "labels" (so the "enter_long_marked" would become your label).


This is very well described in the freqAI documentation - more specifically in the [important key patterns](https://www.freqtrade.io/en/stable/freqai-configuration/#important-dataframe-key-patterns) section.

**Comment by fridary** [comment]
@xmatthias thank you for answer. No, that's not what I mean.
Doing labeling I tell freqtrade features on what data to train. I want to do exactly this, but exclude data that is no labeled from training. Currently freqtrade trains models on all data that is passed through `--timerange ...` param. I want train data within this timerange, but only by **labeled** rows (in my case above labeled rows have `enter_long_marked=1`). Why?  Make experiments with excluding noises. Then, I will enter positions only where `enter_long_marked=1` (algorithmic trading) + checking xgboost predictions on this rows.

**Comment by xmatthias** [comment]
you can probably (abuse) data clensing of freqAI for this by setting the value to np.NaN (instead of 1).

These are usually removed from the training data - so might be working (Though i didn't test if it really works).

I'm not entirely sure it's a good idea - but that's a different topic
it's like "i hide half of the chart from you - now make a good trade" ... but it's up to to evaluate if this method works or not :laughing:
